<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="robot_name" default="4omni"/>
  <group ns="$(arg robot_name)">
    <arg name="model" default="$(find robot_description)/urdf/4omni.urdf"/>   
	  <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>

    <!-- CAN -->
    <node name="usb_can_node" pkg="robot_control" type="usb_can_node" output="screen">
    	<param name="port" value="/dev/ttyUSB0"/>
	<remap from="/can_tx" to="can_tx"/>
	<remap from="/can_rx" to="can_rx"/>
	<remap from="/can_rx" to="can_rx"/>
	<remap from="/can_rx" to="can_rx"/>
	<remap from="/can_rx" to="can_rx"/>
    </node>

    <node name="tr_can" pkg="robot_control" type="tr_can" output="screen">
      <rosparam command="load" file="$(find robot_control)/config/tr_can.yaml" />
		  <remap from="beta/motor0_cmd_vel" to="base/motor0_cmd_vel"/>
		  <remap from="beta/motor1_cmd_vel" to="base/motor1_cmd_vel"/>
		  <remap from="beta/motor2_cmd_vel" to="base/motor2_cmd_vel"/>
		  <remap from="beta/motor3_cmd_vel" to="base/motor3_cmd_vel"/>
    </node>

  </group>
</launch>
